agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationBuilder.java

```diff
@@ -113,6 +118,29 @@ private static void overlayProfilerConfiguration(Configuration config) {
                 .parseBoolean(overlayWithEnvVar(APPLICATIONINSIGHTS_PROFILER_ENABLED, Boolean.toString(config.preview.profiler.enabled)));
     }
 
+    private static void overlayAadConfiguration(Configuration config) {
+        String aadAuthString = getEnvVar(APPLICATIONINSIGHTS_AUTHENTICATION_STRING);
+        if(aadAuthString != null) {
+            Map<String, String> keyValueMap = Splitter.on(";")
+                    .trimResults()
+                    .omitEmptyStrings()
+                    .withKeyValueSeparator("=")
+                    .split(aadAuthString);
+            if(keyValueMap.containsKey("Authorization") && keyValueMap.get("Authorization").equals("AAD")) {
+                // Override any configuration from json
+                config.preview.authentication = new Configuration.AadAuthentication();
+                config.preview.authentication.enabled = true;
+                config.preview.authentication.type = AuthenticationType.SAMI;
+                if(keyValueMap.containsKey("AppId") && !keyValueMap.get("AppId").isEmpty()) {
+                    // Override type to User Assigned Managed Identity
+                    config.preview.authentication.type = AuthenticationType.UAMI;
+                    config.preview.authentication.clientId = keyValueMap.get("AppId");
+                }
```

Comment:
why is it `AppId` in env var, and `ClientId` in json? is there a difference? what's the preferred terminology?

Reply:
updated to use ClientId 👍